soc/arm/st_stm32/stm32u5/soc.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2021 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for STM32U5 processor
+ */
+
+#include <device.h>
+#include <init.h>
+#include <stm32_ll_bus.h>
+#include <stm32_ll_pwr.h>
+#include <stm32_ll_icache.h>
```

Comment:
> If the stm32_ll_icache.h is a new file of ../modules/hal/stm32/stm32cube/common_ll/include
could you link the commit

It's part of cube update for u5:
https://github.com/zephyrproject-rtos/hal_stm32/pull/109/commits/1ef480ba5f556084a072d490969051aa07b1fddf#diff-6a48f59beb44075e26d8e138800b82ffe6cba4004bcad8536d1c159865a05b5b

> else it could use the the SCB_EnableICache() functions

This looks different with what is done in LL_ICACHE_Enable.
I'll dig into that. If similar, I agree that CMSIS functions should be preferred.

Reply:
After investigation, it seems that stm23u5 icache is a dedicated hardware block with specific registers ([Cf Ref man, chapter 8](https://www.st.com/resource/en/reference_manual/rm0456-stm32u575585-armbased-32bit-mcus-stmicroelectronics.pdf)).
So, I'm keeping the proposed implementation